src/frontend/src/pages/FinancePage/CreateReimbursementRequestForm/CreateReimbursementRequestForm.tsx

```diff
@@ -0,0 +1,188 @@
+});
+
+const CreateReimbursementRequestForm = () => {
+  const {
+    handleSubmit,
+    control,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      vendorId: '',
+      account: ClubAccount.CASH,
+      dateOfExpense: new Date(),
+      expenseTypeId: '',
+      reimbursementProducts: [] as ReimbursementProductCreateArgs[],
+      receiptFiles: [] as {
+        file: File;
+      }[]
+    }
+  });
+
+  const {
+    fields: receiptFiles,
+    append: receiptAppend,
+    remove: receiptRemove
+  } = useFieldArray({
+    control,
+    name: 'receiptFiles'
+  });
+  const {
+    fields: reimbursementProducts,
+    append: reimbursementProductAppend,
+    remove: reimbursementProductRemove
+  } = useFieldArray({
+    control,
+    name: 'reimbursementProducts'
+  });
+
+  const { isLoading: createReimbursementRequestIsLoading, mutateAsync: createReimbursementRequest } =
+    useCreateReimbursementRequest();
+  const { isLoading: uploadReceiptsIsLoading, mutateAsync: uploadReceipts } = useUploadManyReceipts();
+  const {
+    isLoading: allVendorsIsLoading,
+    isError: allVendorsIsError,
+    error: allVendorsError,
+    data: allVendors
+  } = useGetAllVendors();
+  const {
+    isLoading: allExpenseTypesIsLoading,
+    isError: allExpenseTypesIsError,
+    error: allExpenseTypesError,
+    data: allExpenseTypes
+  } = useGetAllExpenseTypes();
+
+  const {
+    isLoading: allProjectsIsLoading,
+    isError: allProjectsIsError,
+    error: allProjectsError,
+    data: allProjects
+  } = useAllProjects();
+
+  const toast = useToast();
+  const history = useHistory();
+  const totalCost = reimbursementProducts.reduce((acc, curr) => Number(acc) + Number(curr.cost), 0);
+
+  if (allVendorsIsError) return <ErrorPage message={allVendorsError?.message} />;
+  if (allExpenseTypesIsError) return <ErrorPage message={allExpenseTypesError?.message} />;
+  if (allProjectsIsError) return <ErrorPage message={allProjectsError?.message} />;
+
+  if (
+    allExpenseTypesIsLoading ||
+    allVendorsIsLoading ||
+    allProjectsIsLoading ||
+    createReimbursementRequestIsLoading ||
+    uploadReceiptsIsLoading ||
+    !allVendors ||
+    !allExpenseTypes ||
+    !allProjects
+  )
+    return <LoadingIndicator />;
+
+  const onSubmit = async (data: CreateReimbursementRequestFormInput) => {
+    try {
+      const { reimbursementRequestId } = await createReimbursementRequest({ ...data, totalCost: totalCost });
+      await uploadReceipts({
+        id: reimbursementRequestId,
+        files: data.receiptFiles.map((receiptFile) => receiptFile.file)
+      });
+      history.push(routes.FINANCE + '/' + reimbursementRequestId);
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        toast.error(e.message, 3000);
+      }
+    }
+  };
+
+  const allWbsElements: {
+    wbsNum: WbsNumber;
+    wbsName: string;
+  }[] = allProjects
```

Comment:
Would be good to make this a method in case we end up wanting to use it anywhere else